packages/hash/api/src/graphql/typeDefs/knowledge/org.typedef.ts

```diff
@@ -0,0 +1,38 @@
+import { gql } from "apollo-server-express";
+
+export const persistedOrgTypedef = gql`
+  enum OrgSize {
+    ELEVEN_TO_FIFTY
+    FIFTY_ONE_TO_TWO_HUNDRED_AND_FIFTY
+    ONE_TO_TEN
+    TWO_HUNDRED_AND_FIFTY_PLUS
+  }
+
+  extend type Mutation {
+    """
+    Create an organization.
+    """
+    createOrg(
+      """
+      The shortname of the organization.
+      """
+      shortname: String!
+      """
+      The name of the organization.
+      """
+      name: String!
+      """
+      The size of the organization.
+      """
+      orgSize: OrgSize!
+      """
+      The depth of links that are returned in the response subgraph.
+      """
+      linkResolveDepth: Int
+      """
+      The depth of link target entities that are returned in the response subgraph.
+      """
+      linkTargetEntityResolveDepth: Int
```

Comment:
Where we set defaults we should add them to the schema as makes the mutation better described for consumers.

```suggestion
      """
      linkResolveDepth: Int! = 0
      """
      The depth of link target entities that are returned in the response subgraph.
      """
      linkTargetEntityResolveDepth: Int! = 0
```

By default, the generated types will still have these as nullable, meaning that you still need to handle them being undefined in the resolver (or assert that they're defined). You can change this behaviour in `api/codegen.yml` by adding the following config (last two lines) – this shouldn't be done in clients' codegen config (e.g. `frontend`) because the values _should_ be nullable there, since you don't have to provide them.

```yaml
  ./src/graphql/apiTypes.gen.ts:
    documents:
      - ../shared/src/queries/**/*.ts
    hooks:
      afterOneFileWrite:
        - prettier --write
    plugins:
      - "typescript"
      - "typescript-resolvers"
      - "typescript-operations"
    config:
      avoidOptionals: <-- new
        defaultValue: true <-- new
```

Reply:
I forgot you could set default values in GQL, much prefer this approach. Have addressed in https://github.com/hashintel/hash/pull/1371/commits/a0f2c3312a5509000aec9bd036fc760f9051a651, thanks for flagging :)